app/femr/business/services/system/MissionTripService.java

```diff
@@ -155,6 +155,9 @@ public MissionTripService(IRepository<IMissionCity> missionCityRepository,
     @Override
     public Optional<IMissionTrip> retrieveCurrentMissionTrip(int userId) {
 
+        //TODO should this be moved to just be identified whenever a CurrentUser is retrieved?
+        // This should be contained in the session service??
+
```

Comment:
If this became part of the sessionService/CurrentUser and we wanted to get the current trip for a user that is not logged in (think editing a user as an Administrator and wanting to know this information) then I think we would be out of luck? This doesn't happen right now though.

Otherwise you're right